### PR TITLE
Create TensorBoard test classes in all cases

### PR DESCRIPTION
**Pull request by orionr**
To give better signal to the user, we will now always create the TensorBoard tests classes and just  disable tests if TensorBoard is not installed.

cc @lanpa @sanekmelnikov @natalialunova @pietern 
[test macos]

### CODE REVIEW

**Comment by pietern** [comment]
Can you hold off with merging until #25993 is merged? I'm really close to getting the macOS test run going again and worry this breaks it.

**Comment by orionr** [comment]
Absolutely - land your things first.

**Review by sanekmelnikov — APPROVED** [review]

**Pull request closed without merging**
